device/cuda/src/seeding/doublet_counting.cu

```diff
@@ -24,13 +30,18 @@ __global__ void doublet_counting_kernel(
     const seedfinder_config config, sp_grid_view internal_sp_view,
     doublet_counter_container_view doublet_count_view);
 
-void doublet_counting(const seedfinder_config& config, sp_grid& internal_sp,
-                      host_doublet_counter_container& doublet_counter_container,
+void doublet_counting(const seedfinder_config& config,
+                      sp_grid_view internal_sp_view,
+                      doublet_counter_container_view dcc_view,
                       vecmem::memory_resource& resource) {
 
-    auto doublet_counter_container_view =
-        get_data(doublet_counter_container, &resource);
-    auto internal_sp_view = get_data(internal_sp, resource);
+    unsigned int nbins = internal_sp_view._data_view.m_size;
+
+    // zero initialization
+    set_zero_kernel<<<1, nbins>>>(dcc_view);
```

Comment:
I don't like it that we're maximising `nbins` at 1024. It would be worth to generalise this kernel launch to any bin count.

Reply:
Resolved